src/data/polynomial/derivative.lean

```diff
@@ -87,11 +98,19 @@ by simp [bit1]
   derivative (f + g) = derivative f + derivative g :=
 derivative.map_add f g
 
+@[simp] lemma iterate_derivative_add {f g : polynomial R} {k : ℕ} :
+  derivative^[k] (f + g) = (derivative^[k] f) + (derivative^[k] g) :=
+derivative.to_add_monoid_hom.iterate_map_add _ _ _
+
 @[simp] lemma derivative_neg {R : Type*} [ring R] (f : polynomial R) :
   derivative (-f) = - derivative f :=
 linear_map.map_neg derivative f
 
-@[simp] lemma derivative_sub {R : Type*} [ring R] (f g : polynomial R) :
+@[simp] lemma iterate_derivative_neg {R : Type*} [ring R] {f : polynomial R} {k : ℕ} :
+  derivative^[k] (-f) = - (derivative^[k] f) :=
+(@derivative R _).to_add_monoid_hom.iterate_map_neg _ _
+
+@[simp] lemma derivative_sub {R : Type*} [ring R] {f g : polynomial R} :
```

Comment:
What about an iterative version of this for consistency?

Reply:
Nevermind, I found it, but I don't understand why it needs `comm_ring` and not just `ring`.

Reply:
Nor me! I've reduced it to `ring`.